spectrum-network/src/peer_manager.rs

```diff
@@ -524,7 +528,11 @@ impl<S: PeersState> PeerManagerNotificationsBehavior for PeerManager<S> {
         match self.state.peer(&peer_id) {
             Some(PeerInState::Connected(mut cp)) => {
                 cp.adjust_reputation(ReputationChange::NoResponse);
-                cp.disconnect();
+                if cp.get_reputation() < self.conf.misbehaving_peer_threshold {
```

Comment:
According to the docs you left under `misbehaving_peer_threshold` "misbehaving peer should be dropped", but here you also forget him. Please sync the documentation.

Also why are you checking the threshold in only on dial_failure? I guess PM should check if reputation of a peer fell below the threshold each time it's adjusted (or periodically?) no matter the reason.

Reply:
Changes to reputation now all go through `PeerManagerRequestsBehavior::on_report_peer()`, so on all adjustments. This means we'll need to use `PeerManager::disconnect(..)` whenever a peer's reputation is too low.